### PR TITLE
chore: add yaml linter

### LINKED ISSUE #258: 🛠 Tooling: Add alphabetical sorting to yaml files?
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

#50 and #53 added alphabetical sorting to keys in a couple other forms of code. But right now nothing verifies that `.yaml`/`.yml` files are in alphabetical order. Which, for the sake of keeping things organized & easier to find things in, would be really nice IMO.

### Additional Info

I don't know what the best tool would be - an ESLint plugin? A standalone YAML linter? Investigation required! Before settling on one, I'd like to see what the options are, and what they support.

### PR DESCRIPTION
**Pull request by nowyDEV**
## PR Checklist

- [x] Addresses an existing open issue: fixes #258
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/template-typescript-node-package/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

- Added yaml linter, sorted keys in all yaml files

Screenshot_2023-03-10_09-15-49

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -47,6 +64,48 @@ module.exports = {
...
+							"runs-on",
+							"using",
+							"steps",
+							{
+								order: {
+									natural: false,
+									type: "asc",
+								},
+							},
+						],
```
Comment (JoshuaKGoldberg) [inline .eslintrc.cjs:96]: I would actually rather just get rid of all overrides and be in alphabetical order altogether. Simpler configuration and fewer personal preferences to remember.

File: .eslintignore [hunk 2]
```diff
@@ -1,2 +1,5 @@
 coverage
 lib
+pnpm-lock.yaml
+# Dot files and folders are ignored by default by ESLint
+!.github
```
Comment (JoshuaKGoldberg) [inline .eslintignore:5]: Ah, nice spot. We should also un-exclude `.vscode`. Maybe `!.*` would be a good thing, if it works?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
> multiple issues are inside 1 PR

Indeed - this is a lot of great stuff, but I'll need it split into separate PRs. That way they can be tracked individually and merged as they're ready to go. Could you please?

I'll given this one a round of review to point out things in the meantime. Cheers! ✨

File: .eslintignore (hunk 2 above)
Comment (nowyDEV) [reply .eslintignore]: Added this in https://github.com/JoshuaKGoldberg/template-typescript-node-package/pull/286

File: .eslintignore [hunk 3]
```diff
@@ -1,2 +1,3 @@
 coverage
 lib
+pnpm-lock.yaml
```
Comment (JoshuaKGoldberg) [inline .eslintignore:3]: [Request] Adding `pnpm-lock.yaml` is a good start. But without `!.*` added the `.github/*/*.yml` files aren't getting linted!

```suggestion
!.*
coverage
lib
node_modules
pnpm-lock.yaml
```

In my local test it looks like `.release-it.json` and `.vscode/launch.json` have some discovered out-of-order keys. It's fine to include those in this PR too. Splitting up PRs gets annoying after a while heh.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
The configuration looks great, thanks @nowyDEV! 🙌 

Just requesting changes on the `.eslintignore` so that they actually get applied to files. LMK if you think I'm suggesting something wrong? If not either of us can `pnpm lint --fix` after that suggestion, and then this PR should be good to be merged.

File: .eslintignore (hunk 3 above)
Comment (nowyDEV) [reply .eslintignore]: Done, it does indeed :smile:

Comment (JoshuaKGoldberg) [reply .eslintignore]: Yes please to the new issue, that'd be great!

**Review by JoshuaKGoldberg — APPROVED** [review]
💯 this is great, thanks @nowyDEV! 🙌

**Pull request merged**